src/Parsing/Impl/Ast/ClassDefinition.cs

```diff
@@ -20,31 +20,43 @@
 using System.Threading.Tasks;
 using Microsoft.Python.Core.Collections;
 using Microsoft.Python.Core.Text;
+using Microsoft.Python.Parsing.Definition;
 
 namespace Microsoft.Python.Parsing.Ast {
-    public class ClassDefinition : ScopeStatement {
-        private readonly NameExpression/*!*/ _name;
+    public class ClassDefinition : ScopeStatement, IScopeStatement {
+        private readonly NameExpression _name;
+
         private readonly Statement _body;
         private DecoratorStatement _decorators;
 
-        public ClassDefinition(NameExpression/*!*/ name, ImmutableArray<Arg> bases, Statement body) {
+        public ClassDefinition(NameExpression name, ImmutableArray<Arg> bases, Statement body) {
             _name = name;
             Bases = bases;
             _body = body;
+            ScopeInfo = new ClassScopeInfo(this);
         }
 
         public override int KeywordLength => 5;
 
         public int HeaderIndex { get; set; }
 
-        public override string/*!*/ Name => _name.Name ?? "";
-
-        public NameExpression/*!*/ NameExpression => _name;
+        public NameExpression /*!*/ NameExpression => _name;
 
         public ImmutableArray<Arg> Bases { get; }
 
+        #region IScopeStatement
+
         public override Statement Body => _body;
 
+        #endregion
+
+        #region ScopeStatement
+
+        public override string Name => _name?.Name ?? string.Empty;
```

Comment:
Not: Now that this is restored, you can probably move it back to where it was before. I don't think `Name` is specifically related to `ScopeStatement` and doesn't need to be in this region.

Reply:
Ok